packages/bento-design-system/src/SelectField/BaseSelect.tsx

```diff
@@ -66,81 +72,99 @@ export function BaseSelect<A>(props: Props<A>) {
-          .sort((a, b) => {
-            // In case of multi-select, we display the selected options first
-            if (isMulti) {
-              const selectedValues = (value ?? []) as readonly A[];
-              const isSelected = (a: SelectOption<A>) => selectedValues.includes(a.value);
-              if (isSelected(a) && !isSelected(b)) {
-                return -1;
-              }
-              if (!isSelected(a) && isSelected(b)) {
-                return 1;
-              }
-            }
-            return 0;
-          })}
-        placeholder={placeholder}
-        menuPortalTarget={menuPortalTarget.current}
-        components={selectComponents}
-        openMenuOnFocus={props.openMenuOnFocus ?? dropdownConfig.openMenuOnFocus}
-        styles={selectComponents.styles<SelectOption<A>>()}
-        validationState={validationState}
-        isMulti={isMulti}
-        isClearable={false}
-        noOptionsMessage={() => noOptionsMessage ?? defaultMessages.SelectField.noOptionsMessage}
-        multiValueMessage={
-          props.isMulti && (!props.multiSelectMode || props.multiSelectMode === "summary")
-            ? props.multiValueMessage ?? defaultMessages.SelectField.multiOptionsSelected
-            : undefined
-        }
-        closeMenuOnSelect={!isMulti}
-        hideSelectedOptions={false}
-        menuSize={menuSize}
-        menuIsOpen={isReadOnly ? false : undefined}
-        isSearchable={isReadOnly ? false : searchable ?? true}
-        showMultiSelectBulkActions={isMulti ? props.showMultiSelectBulkActions : false}
-        clearAllButtonLabel={
-          isMulti
-            ? props.clearAllButtonLabel ?? defaultMessages.SelectField.clearAllButtonLabel
-            : undefined
         }
-        selectAllButtonLabel={
-          isMulti
-            ? props.selectAllButtonLabel ?? defaultMessages.SelectField.selectAllButtonLabel
-            : undefined
+        return 0;
+      });
+
+  const commonProps: ComponentProps<Select> = {
+    id: id,
+    instanceId: id,
+    name: name,
+    "aria-label": fieldProps["aria-label"],
+    "aria-labelledby": fieldProps["aria-labelledby"],
+    isDisabled: disabled,
+    isReadOnly: isReadOnly || false,
+    autoFocus: autoFocus,
+    value: assignValue ?? null,
+    onChange: (o) => {
+      if (isMulti) {
+        const multiValue = o as MultiValueT<SelectOption<A>>;
+        onChange(multiValue.map((a) => a.value));
+      } else {
+        const singleValue = o as SingleValueT<SelectOption<A>>;
+        if (clearable) {
+          onChange(singleValue == null ? undefined : singleValue.value);
+        } else {
+          singleValue != null && onChange(singleValue.value);
         }
-        multiSelectMode={isMulti ? props.multiSelectMode : undefined}
-      />
+      }
+    },
+    onBlur: onBlur,
+    options: sortOptions(options),
+    placeholder: placeholder,
+    menuPortalTarget: menuPortalTarget.current,
+    components: selectComponents as any,
```

Comment:
Here and on `styles` I had to use `as any`, and I haven't found a way to make the compiler happy.